Lib/fontTools/misc/psCharStrings.py

```diff
@@ -1031,9 +1032,12 @@ def compile(self, isCFF2=False):
 		self.setBytecode(bytecode)
 
 		if isCFF2:
+			self.isCFF2 = True
```

Comment:
We specifically decided that we don't want `self.isCFF2` state on objects if it can be avoided.  In this case, it can.  Nothing about a charstring makes it a CFF or CFF2 charstring, as they are fully compatible.  It's only during compile that we decide whether it conforms to the resulting format.  Also, compile should not leave a state on the object.  Please rework without it.

Reply:
reworked. Required setting more things in the tests themselves.